File: PTIT123F.cpp
#include<stdio.h>

struct JAIL
{
	int status;	
};

int main(void)
{
	int T;
	scanf("%d",&T);
	while(T--)
	{
		int n;
		scanf("%d",&n);
		JAIL Jail[n+1];
		for(int i=1;i<=n;i++)
			Jail[i].status=1;
		for(int i=2;i<=n;i++)
		{
			for(int j=1;j<=n;j++)
				{
					if(j%i==0)
					{
						if(Jail[j].status==0)
							Jail[j].status=1;
						else
							Jail[j].status=0;
					}
				}
		}
		int count=0;
		for(int i=1;i<=n;i++)
			if(Jail[i].status==1)
				count++;
		printf("%d",count);
		if(T!=0)
			printf("\n");
	}
 return 0;
}

File: spoj py/PTIT123F.py
T=int(input())
while T>0:
    n=int(input())
    Jail=range(0,n+1)
    for i in range(1,n+1):
        Jail[i]=1
    for i in range(2,n+1):
        for j in range(1,n+1):
            if j%i==0:
                if Jail[j]==0:
                    Jail[j]=1
                else:
                    Jail[j]=0
    count=0
    for i in range(1,n+1):
        if Jail[i]==1:
            count=count+1
    print count 
    del Jail[0:n+1]
    del n
    T=T-1
del count
